src/SWP/Component/TemplatesSystem/Gimme/Widget/AbstractWidgetHandler.php

```diff
@@ -88,9 +88,12 @@ public function isVisible()
     /**
      * Render given template with given parameters.
      */
-    protected function renderTemplate($templateName, $parameters = array())
+    protected function renderTemplate($templateName, $parameters = null)
     {
-        $this->container->get('templating')->render(self::WIDGET_TEMPLATE_PATH.'/'.$templateName, $parameters);
+        if (null === $parameters) {
+            $parameters = $this->getAllParametersWithValue();
```

Comment:
in `getModelParameter()` when there is no set or default value you should return `null`.